File: DataTierGeneratorPlusTester/Cs_BuiltIn_WcfServiceApp/Test_masterContract.cs
using System;
using System.Data;
using System.Runtime.Serialization;
using System.ServiceModel;
//using System.ComponentModel;

namespace Cs_BuiltIn_WcfServiceApp 
{
	/// <summary>
	/// Class that defines the table operations.
	/// </summary>
	[ServiceContract]
	public interface ITest_masterService 
	{

		/// <summary>
		/// Inserts a record into the [dbo].[test_master] table.
		/// </summary>
		[OperationContract]
		void Insert(Int32 id, String description, String notes, Int32 someint, Int32? someint_nullable, DateTime somedate, DateTime? somedate_nullable, Double somefloat, Double? somefloat_nullable, Boolean somebool, Boolean? somebool_nullable);

		/// <summary>
		/// Inserts a record into the [dbo].[test_master] table.
		/// </summary>
		[OperationContract]
		void InsertInfo(Test_masterContract info);

		/// <summary>
		/// Updates a record in the [dbo].[test_master] table.
		/// </summary>
		[OperationContract]
		void Update(Int32 id, String description, String notes, Int32 someint, Int32? someint_nullable, DateTime somedate, DateTime? somedate_nullable, Double somefloat, Double? somefloat_nullable, Boolean somebool, Boolean? somebool_nullable);

		/// <summary>
		/// Updates a record in the [dbo].[test_master] table.
		/// </summary>
		[OperationContract]
		void UpdateInfo(Test_masterContract info);

		/// <summary>
		/// Deletes a record from the [dbo].[test_master] table by a composite primary key.
		/// </summary>
		[OperationContract]
		void Delete(Int32 id);

		/// <summary>
		/// Deletes a record from the [dbo].[test_master] table by a composite primary key.
		/// </summary>
		[OperationContract]
		void DeleteInfo(Test_masterContract info);

		/// <summary>
		/// Selects a single record from the [dbo].[test_master] table.
		/// </summary>
		[OperationContract]
		DataSet SelectDS(Int32 id);

		/// <summary>
		/// Selects a single record from the [dbo].[test_master] table.
		/// </summary>
		[OperationContract]
		BindingListView<Test_masterContract> SelectInfo(Test_masterContract info);

		/// <summary>
		/// Selects all records from the [dbo].[test_master] table.
		/// </summary>
		[OperationContract]
		DataSet SelectDSAll();

		/// <summary>
		/// Selects all records from the [dbo].[test_master] table.
		/// </summary>
		[OperationContract]
		BindingListView<Test_masterContract> SelectInfoAll();
	}

	/// <summary>
	/// Class that stores table fields.
	/// </summary>
	[DataContract]
	public class Test_masterContract 
	{
		/// <summary>
		/// Default constructor.  
		/// </summary>
		public Test_masterContract() {}

		/// <summary>
		/// Constructor with values.  
		/// </summary>
		public Test_masterContract(
			Int32 id,
			String description,
			String notes,
			Int32 someint,
			Int32? someint_nullable,
			DateTime somedate,
			DateTime? somedate_nullable,
			Double somefloat,
			Double? somefloat_nullable,
			Boolean somebool,
			Boolean? somebool_nullable
		)
		{
			_id = id;
			_description = description;
			_notes = notes;
			_someint = someint;
			_someint_nullable = someint_nullable;
			_somedate = somedate;
			_somedate_nullable = somedate_nullable;
			_somefloat = somefloat;
			_somefloat_nullable = somefloat_nullable;
			_somebool = somebool;
			_somebool_nullable = somebool_nullable;
		}

		/// Private variables for columns in the test_master table.
		private Int32 _id;
		private String _description;
		private String _notes;
		private Int32 _someint;
		private Int32? _someint_nullable;
		private DateTime _somedate;
		private DateTime? _somedate_nullable;
		private Double _somefloat;
		private Double? _somefloat_nullable;
		private Boolean _somebool;
		private Boolean? _somebool_nullable;

		/// Public properties for columns in the test_master table.

		/// <summary>
		/// id  
		/// </summary>
		[DataMember]
		//[DataObjectFieldAttribute(true, false, false)]
		public Int32 id
		{
			get { return _id; }
			set { _id = value; }
		}

		/// <summary>
		/// description  
		/// </summary>
		[DataMember]
		//[DataObjectFieldAttribute(false, false, false)]
		public String description
		{
			get { return _description; }
			set { _description = value; }
		}

		/// <summary>
		/// notes  
		/// </summary>
		[DataMember]
		//[DataObjectFieldAttribute(false, false, true)]
		public String notes
		{
			get { return _notes; }
			set { _notes = value; }
		}

		/// <summary>
		/// someint  
		/// </summary>
		[DataMember]
		//[DataObjectFieldAttribute(false, false, false)]
		public Int32 someint
		{
			get { return _someint; }
			set { _someint = value; }
		}

		/// <summary>
		/// someint_nullable  
		/// </summary>
		[DataMember]
		//[DataObjectFieldAttribute(false, false, true)]
		public Int32? someint_nullable
		{
			get { return _someint_nullable; }
			set { _someint_nullable = value; }
		}

		/// <summary>
		/// somedate  
		/// </summary>
		[DataMember]
		//[DataObjectFieldAttribute(false, false, false)]
		public DateTime somedate
		{
			get { return _somedate; }
			set { _somedate = value; }
		}

		/// <summary>
		/// somedate_nullable  
		/// </summary>
		[DataMember]
		//[DataObjectFieldAttribute(false, false, true)]
		public DateTime? somedate_nullable
		{
			get { return _somedate_nullable; }
			set { _somedate_nullable = value; }
		}

		/// <summary>
		/// somefloat  
		/// </summary>
		[DataMember]
		//[DataObjectFieldAttribute(false, false, false)]
		public Double somefloat
		{
			get { return _somefloat; }
			set { _somefloat = value; }
		}

		/// <summary>
		/// somefloat_nullable  
		/// </summary>
		[DataMember]
		//[DataObjectFieldAttribute(false, false, true)]
		public Double? somefloat_nullable
		{
			get { return _somefloat_nullable; }
			set { _somefloat_nullable = value; }
		}

		/// <summary>
		/// somebool  
		/// </summary>
		[DataMember]
		//[DataObjectFieldAttribute(false, false, false)]
		public Boolean somebool
		{
			get { return _somebool; }
			set { _somebool = value; }
		}

		/// <summary>
		/// somebool_nullable  
		/// </summary>
		[DataMember]
		//[DataObjectFieldAttribute(false, false, true)]
		public Boolean? somebool_nullable
		{
			get { return _somebool_nullable; }
			set { _somebool_nullable = value; }
		}
	}
}

File: DataTierGeneratorPlusTester/Vb_BuiltIn_WcfServiceApp/Test_masterContract.vb
Imports System
Imports System.Data
Imports System.Runtime.Serialization
Imports System.ServiceModel
'Imports System.ComponentModel 

'Namespace Vb_BuiltIn_WcfServiceApp 
	''' <summary>
	''' Class that defines the table operations.
	''' </summary>
	<ServiceContract()> _
	Public Interface ITest_masterService 

		''' <summary>
		''' Inserts a record into the [dbo].[test_master] table.
		''' </summary>
		<OperationContract()> _
		Sub Insert(id As Int32, description As String, notes As String, someint As Int32, someint_nullable As Int32?, somedate As DateTime, somedate_nullable As DateTime?, somefloat As Double, somefloat_nullable As Double?, somebool As Boolean, somebool_nullable As Boolean?)

		''' <summary>
		''' Inserts a record into the [dbo].[test_master] table.
		''' </summary>
		<OperationContract()> _
		Sub InsertInfo(ByVal info As Test_masterContract)

		''' <summary>
		''' Updates a record in the [dbo].[test_master] table.
		''' </summary>
		<OperationContract()> _
		Sub Update(id As Int32, description As String, notes As String, someint As Int32, someint_nullable As Int32?, somedate As DateTime, somedate_nullable As DateTime?, somefloat As Double, somefloat_nullable As Double?, somebool As Boolean, somebool_nullable As Boolean?)

		''' <summary>
		''' Updates a record in the [dbo].[test_master] table.
		''' </summary>
		<OperationContract()> _
		Sub UpdateInfo(ByVal info As Test_masterContract)

		''' <summary>
		''' Deletes a record from the [dbo].[test_master] table by a composite primary key.
		''' </summary>
		<OperationContract()> _
		Sub Delete(id As Int32)

		''' <summary>
		''' Deletes a record from the [dbo].[test_master] table by a composite primary key.
		''' </summary>
		<OperationContract()> _
		Sub DeleteInfo(ByVal info As Test_masterContract)

		''' <summary>
		''' Selects a single record from the [dbo].[test_master] table.
		''' </summary>
		<OperationContract()> _
		Function SelectDS(id As Int32) As DataSet 

		''' <summary>
		''' Selects a single record from the [dbo].[test_master] table.
		''' </summary>
		<OperationContract()> _
		Function SelectInfo(ByVal info As Test_masterContract) As BindingListView(Of Test_masterContract) 

		''' <summary>
		''' Selects all records from the [dbo].[test_master] table.
		''' </summary>
		<OperationContract()> _
		Function SelectDSAll() As DataSet  

		''' <summary>
		''' Selects all records from the [dbo].[test_master] table.
		''' </summary>
		<OperationContract()> _
		Function SelectInfoAll() As BindingListView(Of Test_masterContract)  
	End Interface

	''' <summary>
	''' Class that stores table fields.
	''' </summary>
	<DataContract()> _
	Public Class Test_masterContract 
		''' <summary>
		''' Default constructor.  
		''' </summary>
		Public Sub New() 
		End Sub 

		''' <summary>
		''' Constructor with values.  
		''' </summary>
		Public Sub New( _
			id As Int32, _
			description As String, _
			notes As String, _
			someint As Int32, _
			someint_nullable As Int32?, _
			somedate As DateTime, _
			somedate_nullable As DateTime?, _
			somefloat As Double, _
			somefloat_nullable As Double?, _
			somebool As Boolean, _
			somebool_nullable As Boolean? _
		)
			_id = id
			_description = description
			_notes = notes
			_someint = someint
			_someint_nullable = someint_nullable
			_somedate = somedate
			_somedate_nullable = somedate_nullable
			_somefloat = somefloat
			_somefloat_nullable = somefloat_nullable
			_somebool = somebool
			_somebool_nullable = somebool_nullable
		End Sub

		' Private variables for columns in the test_master table.
		Private  _id As Int32
		Private  _description As String
		Private  _notes As String
		Private  _someint As Int32
		Private  _someint_nullable As Int32?
		Private  _somedate As DateTime
		Private  _somedate_nullable As DateTime?
		Private  _somefloat As Double
		Private  _somefloat_nullable As Double?
		Private  _somebool As Boolean
		Private  _somebool_nullable As Boolean?

		' Public properties for columns in the test_master table.

		'<DataObjectFieldAttribute(True, False, False)> _
		''' <summary>
		''' id  
		''' </summary>
		<DataMember()> _
		Public Property id As Int32
			Get 
				Return _id 
			End Get
			Set(ByVal Value As Int32) 
				_id = Value
			End Set
		End Property

		'<DataObjectFieldAttribute(False, False, False)> _
		''' <summary>
		''' description  
		''' </summary>
		<DataMember()> _
		Public Property description As String
			Get 
				Return _description 
			End Get
			Set(ByVal Value As String) 
				_description = Value
			End Set
		End Property

		'<DataObjectFieldAttribute(False, False, True)> _
		''' <summary>
		''' notes  
		''' </summary>
		<DataMember()> _
		Public Property notes As String
			Get 
				Return _notes 
			End Get
			Set(ByVal Value As String) 
				_notes = Value
			End Set
		End Property

		'<DataObjectFieldAttribute(False, False, False)> _
		''' <summary>
		''' someint  
		''' </summary>
		<DataMember()> _
		Public Property someint As Int32
			Get 
				Return _someint 
			End Get
			Set(ByVal Value As Int32) 
				_someint = Value
			End Set
		End Property

		'<DataObjectFieldAttribute(False, False, True)> _
		''' <summary>
		''' someint_nullable  
		''' </summary>
		<DataMember()> _
		Public Property someint_nullable As Int32?
			Get 
				Return _someint_nullable 
			End Get
			Set(ByVal Value As Int32?) 
				_someint_nullable = Value
			End Set
		End Property

		'<DataObjectFieldAttribute(False, False, False)> _
		''' <summary>
		''' somedate  
		''' </summary>
		<DataMember()> _
		Public Property somedate As DateTime
			Get 
				Return _somedate 
			End Get
			Set(ByVal Value As DateTime) 
				_somedate = Value
			End Set
		End Property

		'<DataObjectFieldAttribute(False, False, True)> _
		''' <summary>
		''' somedate_nullable  
		''' </summary>
		<DataMember()> _
		Public Property somedate_nullable As DateTime?
			Get 
				Return _somedate_nullable 
			End Get
			Set(ByVal Value As DateTime?) 
				_somedate_nullable = Value
			End Set
		End Property

		'<DataObjectFieldAttribute(False, False, False)> _
		''' <summary>
		''' somefloat  
		''' </summary>
		<DataMember()> _
		Public Property somefloat As Double
			Get 
				Return _somefloat 
			End Get
			Set(ByVal Value As Double) 
				_somefloat = Value
			End Set
		End Property

		'<DataObjectFieldAttribute(False, False, True)> _
		''' <summary>
		''' somefloat_nullable  
		''' </summary>
		<DataMember()> _
		Public Property somefloat_nullable As Double?
			Get 
				Return _somefloat_nullable 
			End Get
			Set(ByVal Value As Double?) 
				_somefloat_nullable = Value
			End Set
		End Property

		'<DataObjectFieldAttribute(False, False, False)> _
		''' <summary>
		''' somebool  
		''' </summary>
		<DataMember()> _
		Public Property somebool As Boolean
			Get 
				Return _somebool 
			End Get
			Set(ByVal Value As Boolean) 
				_somebool = Value
			End Set
		End Property

		'<DataObjectFieldAttribute(False, False, True)> _
		''' <summary>
		''' somebool_nullable  
		''' </summary>
		<DataMember()> _
		Public Property somebool_nullable As Boolean?
			Get 
				Return _somebool_nullable 
			End Get
			Set(ByVal Value As Boolean?) 
				_somebool_nullable = Value
			End Set
		End Property
	End Class
'End Namespace
